packages/@aws-cdk/cloudformation-diff/lib/diff/util.ts

```diff
@@ -140,17 +140,21 @@ export function unionOf(lv: string[] | Set<string>, rv: string[] | Set<string>):
  * (for which JavaScript's parseFloat() returns 0).
  */
 function safeParseFloat(str: string): number {
+  const decimalNumberRegex = /^\d*(\.\d*)?$/;
```

Comment:
According to the documentation of `parseFloat`:

> Consider [Number(value)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number) for stricter parsing, which converts to NaN for arguments with invalid characters anywhere.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseFloat

Maybe do that instead?

Reply:
Ah, JavaScript, you will always surprise me, no matter how long I use you 😃.

Will change!